filter/filter.go

```diff
@@ -1084,3 +1084,81 @@ func (c *Client) doPatchWithAuthHeaders(ctx context.Context, userAuthToken, serv
 	// do the request
 	return c.hcCli.Client.Do(ctx, req)
 }
+
+// CreateFilter creates a filter and returns the associated filterID and eTag
+func (c *Client) CreateFilter(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, collectionID, datasetID, edition, version, populationType string, names []string) (filterID, eTag string, err error) {
+	ver, err := strconv.Atoi(version)
+	if err != nil {
+		return "", "", err
+	}
+
+	var dimensions []Dimension
+	for _, name := range names {
+		dimensions = append(dimensions, Dimension{Name: name})
+	}
+
+	filter := CreateFilterRequest{
+		Dataset: Dataset{
+			DatasetID: datasetID,
+			Edition:   edition,
+			Version:   ver,
+		},
+		Dimensions:     dimensions,
+		PopulationType: populationType,
+	}
+
+	data, err := json.Marshal(filter)
+	if err != nil {
+		return "", "", err
+	}
+	uri := c.hcCli.URL + "/filters"
+	clientlog.Do(ctx, "attempting to create filter", service, uri, log.Data{
+		"method":    "POST",
+		"datasetID": datasetID,
+		"edition":   edition,
+		"version":   version,
```

Comment:
We should also log out the population-type here

Reply:
done